src/interop/js-interop/usage.md

```diff
@@ -0,0 +1,368 @@
+---
+title: Usage
+description: How to declare and use JS interop members.
+---
+
+JS interop works by looking up properties in the [global JS scope]. In order to
+interact with these properties, you use [`external`] interop members. In order
+to provide types for these properties, you use interop types.
+
+## Interop types
+
+When interacting with a JS value, you need to provide a Dart type for it. In
+order to do this, you either use or declare an interop type. Dart provides a set
+of core interop types called ["JS types"] that you can use to add types to your
+interop members. You can also declare an interop type using [extension types] by
+making its representation type another interop type. "JS types" refer to the
+interop types provided by Dart, and while all JS types are interop types, not
+all interop types are JS types. For example:
+
+```dart
+extension type Console(JSObject _) implements JSObject {}
+```
+
+Both `Console` and `JSObject` are interop types, but only `JSObject` is a JS
+type.
+
+You can also use `Console` as a representation type of an interop type as it is
+an interop type:
+
+```dart
+extension type Console2(Console _) implements Console {}
+```
+
+Because you're using extension types, both `Console` and `Console2` are still
+`JSObject`s. You're just providing a Dart interface to view the `JSObject`
+differently. There is no guarantee that they are a specific type of JS object.
+
+Generally, you will likely use `JSObject` as the representation type for interop
+types you declare because you're likely interacting with JS objects who don't
+have a corresponding JS type like [`Window`].
+
+Interop types should also generally implement their representation type so that
+they can be subtypes of the representation type.
+
+## Interop members
+
+In order to get a JS value that you can provide a type for, you'll want to use
+an `external` interop member. These members might take in arguments and return a
+type. The types that can be used on these members have [restrictions]. There are
+various ways you can declare interop members.
+
+### Top-level interop members
+
+```dart
+@JS()
+external Console get console;
+
+@JS()
+external set name(String value);
+
+@JS()
+external void close();
+```
+
+Here, `console`, `name`, and `close` are top-level interop members. When
+`console` is called, the value in the `console` property in the global namespace
+is retrieved and casted to `Console`. When `name` is set, the property `name` in
+the global namespace is set to the given value. When `close` is called, the
+function `close` in the global namespace is called with no arguments. You can
+declare top-level interop getters, setters, methods, and fields, which are just
+pairs of getters and setters.
+
+### Interop type members
+
+```dart
+extension type Array._(JSObject _) implements JSObject {
```

Comment:
This example is really nicely explained, I am however wondering if it would be worth morphing it into an example that uses a JavaScript class, rather than a browser type, as the underlying JavaScript API.

The one downside of that is that I can imagine it may be too tricky to include the `operator []` as part of the example, but maybe that's OK?

My motivation is to make the connection between Dart-JS more direct by showing both the original JavaScript and the corresponding Dart interface.